Plutarch/Api/V1/Value.hs

```diff
@@ -502,3 +405,11 @@ zeroData = pdata 0
 pmapAmounts :: Term s ((PInteger :--> PInteger) :--> PValue k a :--> PValue k 'NoGuarantees)
 pmapAmounts = phoistAcyclic $
   plam $ \f v -> pcon $ PValue $ AssocMap.pmap # plam (AssocMap.pmap # f #) # pto v
+
+{- | Given an amount comparison function, check whether a binary relation holds over
+2 sorted 'PValue's.
+-}
+pcheckBinRel :: Term s ((PInteger :--> PInteger :--> PBool) :--> PValue 'Sorted any0 :--> PValue 'Sorted any1 :--> PBool)
+pcheckBinRel = phoistAcyclic $
+  plam $ \f v1 v2 ->
+    AssocMap.pcheckBinRel # (AssocMap.pcheckBinRel # f # 0) # AssocMap.pempty # pto v1 # pto v2
```

Comment:
It probably won't, because the semantics might be different. We don't know whether the inner function would err after 3, 2, or 1 applications. η-reduction only happens when the inner function is a lambda abstraction. That is probably not the case here, so you should η-reduce it manually.